packages/lavamoat-node/src/cli.js

```diff
@@ -24,15 +24,24 @@ function parseArgs() {
       })
       yargsFlags(yargs, defaults)
     })
+    .parserConfiguration({
+      'populate--': true,
+    })
     .help()
 
   const parsedArgs = argsParser.parse()
 
-  // patch process.argv so it matches the normal pattern
+  // Rebuild process.argv to pass only entry args
+  const entryArgs = parsedArgs['--']
+  if (!entryArgs) {
+    // patch process.argv so it matches the normal pattern
   // e.g. [runtime path, entrypoint, ...args]
   // we'll use the LavaMoat path as the runtime
   // so we just remove the node path
   process.argv.shift()
+  } else {
+    process.argv = [process.argv[0], parsedArgs.entryPath, ...entryArgs]
```

Comment:
```suggestion
    // Rebuild process.argv to pass only entry args
    process.argv = [process.argv[0], parsedArgs.entryPath, ...entryArgs]
```
Nit: More readable to place this inline IMO